proctor-common/src/main/java/com/indeed/proctor/common/RuleEvaluator.java

```diff
@@ -153,6 +157,50 @@ public boolean evaluateBooleanRuleWithValueExpr(final String rule, @Nonnull fina
                 + " from rule " + rule);
     }
 
+    /**
+     * @deprecated Use evaluateBooleanRulePartialWithValueExpr(String, Map) instead, it's more efficient
+     */
+    @Deprecated
+    public boolean evaluateBooleanRulePartial(final String rule, @Nonnull final Map<String, Object> values) throws IllegalArgumentException {
+        final Map<String, ValueExpression> localContext = ProctorUtils.convertToValueExpressionMap(expressionFactory, values);
+        return evaluateBooleanRulePartialWithValueExpr(rule, localContext);
+    }
+
+    public boolean evaluateBooleanRulePartialWithValueExpr(final String rule, @Nonnull final Map<String, ValueExpression> values) throws IllegalArgumentException {
```

Comment:
Where will this be used?

Reply:
Planning to use it at least for implementing a tool to report on what tests are adding what match providers for our recommendation systems. We're dynamically loading proctor tests with payloads that add or remove match providers from the pool, but there are over 200 allocations across all the tests so it's hard to manually look through the allocations and accompanying payloads to figure out how things pan out. We have other cases where it would be helpful to filter allocations to those that could match a partial context.